src/backend/query_builder.rs

```diff
@@ -1512,3 +1513,45 @@ impl QuotedBuilder for CommonSqlQueryBuilder {
 impl EscapeBuilder for CommonSqlQueryBuilder {}
 
 impl TableRefBuilder for CommonSqlQueryBuilder {}
+
+pub(crate) fn common_inner_expr_well_known_greater_precedence(
+    inner: &SimpleExpr,
+    outer_oper: &Oper,
+) -> bool {
+    match inner {
+        SimpleExpr::Column(_) => true,
```

Comment:
We could merge the trivial arms together, and maybe add why they are higher precedence.
I'm not sure I understand why e.g. FunctionCall is higher precedence, what does that mean in practice?

Reply:
Merged arms together and added this comment: 

// We only consider the case where an inner expression is contained in either a
// unary or binary expression (with an outer_oper).
// We do not need to wrap with parentheses:
// Columns, tuples (already wrapped), constants, function calls, values,
// keywords, subqueries (already wrapped)

I dropped SimpleEnum::AsEnum from this list, and the code.
In fact, it looks like there is a bug in sea-query wrt. how these are to be written to SQL?
This is from query_builder.rs:

            SimpleExpr::AsEnum(_, expr) => {
                self.prepare_simple_expr(expr, sql);
            }

The query builder for Postgres has different behaviour:

SimpleExpr::AsEnum(type_name, expr) => {
                let simple_expr = expr.clone().cast_as(SeaRc::clone(type_name));
                self.prepare_simple_expr_common(&simple_expr, sql);
            }

But we can follow that up in an issue.